Custom/Mails/mailhistoryunit.h

```diff
@@ -0,0 +1,97 @@
+#ifndef MAILHISTORYUNIT_H
+#define MAILHISTORYUNIT_H
+
+#include <QWidget>
+#include <qdatetime.h>
+
+namespace Ui {
+class MailHistoryUnit;
+}
+
+struct LetterStruct
+{
+    LetterStruct(){}
+
+    LetterStruct(const QString& _sender, const QString& _recipient
+                 , const QDate& _timestamp, const QString& _subject
+                 , const QString& _body, const QVector<QString> _files_paths)
+        : sender(_sender)
+        , recipient(_recipient)
+        , timestamp(_timestamp)
+        , subject(_subject)
+        , body(_body)
+        , files_paths(_files_paths)
+    {}
+
+    QString sender{};
+    QString recipient{};
+    QDate timestamp{};
+    QString subject{};
+    QString body{};
+    QVector<QString> files_paths{};
+
+    inline QString GenerateFileName() const
+    {
+        return QString(sender + "_" + recipient + "_" + timestamp.toString("dd-MM-yyyy") + ".txt");
+    }
+
+    operator QString() const
+    {
+        QString attached_files{};
+        if (!files_paths.isEmpty())
+        {
+            attached_files = "\n\nAttached files: \n";
+            attached_files += files_paths.join(";\n");
+        }
+        return "From: " + sender + "\nTo: " + recipient + "\nSent at: " + timestamp.toString()
+               + "\nSubject: " + subject + "\n\n" + body + attached_files;
+    }
+
+    friend QDataStream &operator<<(QDataStream &out, const LetterStruct &myStruct)
```

Comment:
code convention: ampersand should be adjacent to the type of variable

Reply:
thanks for noticing, will be fixed!